component/core/src/main/java/org/exoplatform/social/core/storage/cache/CachedRelationshipStorage.java

```diff
@@ -194,6 +196,22 @@ private SuggestionsData buildIdMap(Map<Identity, Integer> map) {
 
   }
 
+  /**
+   * Remove identity and profile cache.
+   *
+   * @param identity
+   */
+  private void removeIdentityCache(Identity identity) {
+    CachedIdentityStorage cachedIdentityStorage =
+                                                (CachedIdentityStorage) PortalContainer.getInstance()
+                                                                                       .getComponentInstanceOfType(IdentityStorage.class);
+    Identity cachedIdentity = cachedIdentityStorage.findIdentity(OrganizationIdentityProvider.NAME, identity.getRemoteId());
+
+    if (cachedIdentity != null) {
+      cachedIdentityStorage.clearIdentityCached(cachedIdentity, identity.getRemoteId());
```

Comment:
It's better to not clear the whole identities cache when the relationship changes between two users. In fact the method `cachedIdentityStorage.clearIdentityCached` will delete all user identities from cache. The clear operation is useful when the identity itself changes, but for relationship, you can just :
1. update the cache time of cached identity only
2. clear only the identity from cache

To do this, I suggest to add a new method similar to clearIdentityCached to clear the cached identity only:
```java
  public void clearIdentityCache(Identity identity, boolean clearList) {
    IdentityKey key = new IdentityKey(new Identity(identity.getId()));
    exoIdentityCache.remove(key);
    IdentityCompositeKey compositeKey = new IdentityCompositeKey(identity.getProviderId(), identity.getRemoteId());
    if (exoIdentityIndexCache.get(compositeKey) != null) {
      exoIdentityIndexCache.remove(compositeKey);
    }
    if (clearList) {
      clearCache();
    }
  }
```

Reply:
Updated!